File: src/main/java/com/dh/ondot/member/api/response/UpdateHomeAddressResponse.java
package com.dh.ondot.member.api.response;

import com.dh.ondot.member.domain.Address;

public record UpdateHomeAddressResponse(
        String roadAddress,
        Double longitude,
        Double latitude
) {
    public static UpdateHomeAddressResponse from(Address address) {
        return new UpdateHomeAddressResponse(
                address.getRoadAddress(),
                address.getLongitude(),
                address.getLatitude()
        );
    }
}

File: src/main/kotlin/com/dh/ondot/member/presentation/response/UpdateHomeAddressResponse.kt
package com.dh.ondot.member.presentation.response

import com.dh.ondot.member.domain.Address

data class UpdateHomeAddressResponse(
    val roadAddress: String,
    val longitude: Double,
    val latitude: Double,
) {
    companion object {
        fun from(address: Address): UpdateHomeAddressResponse =
            UpdateHomeAddressResponse(
                roadAddress = address.roadAddress,
                longitude = address.longitude,
                latitude = address.latitude,
            )
    }
}
